### PR TITLE
TME-2307: add support for existing trail with x-account resources

### PR DESCRIPTION
**Pull request by sarjumulmi**
> Add support for existing cloudtrail with resources deployed in different aws accounts.

Changes:
- `configuration_alias` added for log bucket account. need to pass in the provider alias for log bucket when invoking the module
- conditionally create resources based on `is_existing_cloudtrail_cross_account` var
- local runtime check for assuring other dependent optional variables are provided when `is_existing_cloudtrail_cross_account` is set to true.
- new variable for `expel assume role name` instead of hard coded string. add [type defn](https://github.com/expel-io/terraform-aws-cloudtrail/issues) for `tags`
- change circle ci config as a workaround for `terraform validate` [bug](https://github.com/hashicorp/terraform/issues/28490#issuecomment-1140007955)

### CODE REVIEW

File: examples/existing-cloudtrail/cross-account/terraform.tfvars
Comment (bachu20) [inline examples/existing-cloudtrail/cross-account/terraform.tfvars:5]: Nit:
```suggestion
existing_cloudtrail_log_bucket_account_id = "Replace with your AWS CloudTrail log bucket account id"
```

File: variables.tf
Comment (bachu20) [inline variables.tf:15]: Sweet, thanks for knocking this out too!

**Review by bachu20 — DISMISSED** [review]
lgtm. Would definitely defer to @davidmontoyago + @ethanmil for reviews before merging it in.

File: README.md
Comment (sarjumulmi) [inline README.md:70]: this applies to cases where you have an existing organization cloudtrail but the cloudtrail log bucket account is NOT in the management account.

File: variables.tf
Comment (ethanmil) [inline variables.tf:73]: mgmt -> management

Comment (ethanmil) [reply variables.tf:73]: omit `customer's`

File: variables.tf
Comment (ethanmil) [inline variables.tf:82]: omit `customer's`

**Review by ethanmil — CHANGES_REQUESTED** [review]
Looking good! Just a few typos and readability suggestions.

**Review by davidmontoyago — COMMENTED** [review]
pair-reviewed with @ethanmil

File: .circleci/config.yml
Comment (davidmontoyago) [inline .circleci/config.yml:101]: 💯

**Review by davidmontoyago — APPROVED** [review]
LGTM

**Review by ethanmil — APPROVED** [review]
LGTM

**Review by davidmontoyago — APPROVED** [review]
additions 372aeda and 1a335276dd7b3ed7dfc31616629a44986dc3e22d LGTM

**Pull request merged**
